apps/modernization-ui/src/apps/page-builder/page/management/edit/tabs/ManageTabs/ManageTabs.tsx

```diff
@@ -88,6 +92,44 @@ export const ManageTabs = ({ pageId, onAddSuccess, tabs }: Props) => {
         }
     };
 
+    const handleDelete = async () => {
+        if (selectedForDelete && onAddSuccess) {
+            if (selectedForDelete.sections.length > 0) {
+                setMessage({
+                    type: 'error',
+                    expiration: 3000,
+                    message: (
+                        <p>
+                            This tab cannot be deleted because there are other elements inside it. Please remove or
+                            delete all child elements by managing the content at the subsection/section/question level.
+                        </p>
+                    )
+                });
+                setSelectedForDelete(undefined);
+            } else {
+                await deleteTab(pageId, selectedForDelete.id)
```

Comment:
so you're doing a combination of async await and .then() .catch(). should just stick to one or the other.